HousingSearchApi.Tests/V1/Factories/TenureQueryGeneratorTests.cs

```diff
@@ -0,0 +1,165 @@
+
+            // Act
+            _sut.Create(_request, _queryContainerDescriptor);
+
+            //Assert
+            _queryFilterBuilderMock.Verify();
+        }
+
+        [Fact]
+        public void CreateGeneratesCorrectQueryWhenUsingWildstarQuery()
+        {
+            // Arrange
+            var expectedMultipleFilterQueryFields = new List<string>()
+            {
+                 "tenuredAsset.isTemporaryAccommodation"
+            };
+
+            var expectedWithFilterQueryFields = new List<string>()
+            {
+                "tempAccommodationInfo.bookingStatus"
+            };
+
+            var expectedWildstarFields = new List<string>()
+            {
+                "paymentReference",
+                "tenuredAsset.fullAddress^3",
+                "householdMembers",
+                "householdMembers.fullName^3"
+            };
+
+            var expectedTextQueryType = TextQueryType.MostFields;
+
+            _queryFilterBuilderMock.Setup(x =>
+                x.WithMultipleFilterQuery(It.IsAny<string>(), It.IsAny<List<string>>())
+                .WithFilterQuery(It.IsAny<string>(), It.IsAny<List<string>>(), It.IsAny<TextQueryType>())
+                .WithWildstarQuery(It.IsAny<string>(), It.IsAny<List<string>>(), It.IsAny<TextQueryType>())
+                .Build(It.IsAny<QueryContainerDescriptor<QueryableTenure>>()));
+
+            // Act
+            _ = _sut.Create(_request, _queryContainerDescriptor);
+
+            // Assert
+            _queryFilterBuilderMock.Verify(x =>
+                x.WithMultipleFilterQuery(_request.IsTemporaryAccommodation, expectedMultipleFilterQueryFields)
+                .WithFilterQuery(_request.BookingStatus, expectedWithFilterQueryFields, It.IsAny<TextQueryType>())
+                .WithWildstarQuery(_request.SearchText, expectedWildstarFields, expectedTextQueryType).Build(_queryContainerDescriptor), Times.Once);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(null)]
+        public void CreateGeneratesCorrectQueryWhenUsingExactQuery(string searchText)
+        {
+            // Arrange
+            _request.SearchText = searchText;
+
+            var expectedExactQueryFields = new List<string>()
+            {
+                "tenuredAsset.uprn"
+            };
+
+            var expectedTextQueryType = TextQueryType.MostFields;
+
+            _queryBuilderMock.Setup(x =>
+                x.WithExactQuery(It.IsAny<string>(), It.IsAny<List<string>>(), It.IsAny<IExactSearchQuerystringProcessor>(), It.IsAny<TextQueryType>())
+                .Build(It.IsAny<QueryContainerDescriptor<QueryableTenure>>()));
+
+            // Act
+            _ = _sut.Create(_request, _queryContainerDescriptor);
+
+            // Assert
+            _queryBuilderMock.Verify(x =>
+                x.WithExactQuery(_request.Uprn, expectedExactQueryFields, null, expectedTextQueryType)
+                .Build(_queryContainerDescriptor), Times.Once);
+        }
```

Comment:
Could be handy Verifying that the other query builder methods from the other branch didn't get called.
So `.WithMultipleFilterQuery`, `.WithFilterQuery`, `.WithWildstarQuery` were called `Times.Never`.

Reply:
I think that's worth having, so have [added it](https://github.com/LBHackney-IT/housing-search-api/pull/222/commits/779a705e93d0fbd73d8a0eda5340476b54d67c54) 👍